src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -176,6 +177,10 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 {
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
+                else if (string.IsNullOrEmpty(functionName) && exception is FunctionListenerException flex)
```

Comment:
Let's generalize this to the base exception FunctionException. Currently both FunctionListenerException and FunctionIndexingException are subclasses, so we'll get your improvements for more cases :)

Reply:
That was the initial idea but then I've found that `MethodName` in the `FunctionListenerException` could come as `Functions.{functionName}` even if in the WebJobs Host we are using the `FunctionDescriptor.ShortName`. Then in `FunctionInvocationException` we're removing _Host.Functions_ but not using the short name. Since I didn't find a common normalization for both, I went this way.

Please let me know if you have a better idea on this, if you prefer to go with the base exception or let the _Functions._ in the FunctionName for these scenarios...

Thanks!

Reply:
So there's actually 3 subclasses of this exception type then. I think we should just handle the base exception type and use whatever information we have - some information is better than nothing, right? If we really need to normalize the name at the Functions layer we could do that. Is your concern how detectors would deal with the different formats?

Reply:
I'm aware of FunctionListenerException and FunctionInvocationException extending FunctionExtension. I was looking to maintain consistency, as in other places we log the short name without _Functions._. I'm good with handling FunctionException here and avoid checking for the individual type.